### PR TITLE
Best Practices

### PR DESCRIPTION
**Pull request by abhishek-dhanraj-TI**

### CODE REVIEW

File: restapi/utils.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+# Set baseConfig level to INFO to log all messages except the DEBUG ones
+logging.basicConfig(
+    level=logging.INFO,
+    filename='general.log',
+    filemode='w',
+    format="%(levelname)s :: %(asctime)s :: %(message)s",
+)
+
+logger = logging.getLogger(__name__)
```
Comment (kushal-ti) [inline restapi/utils.py:13]: one slight change. If you're configuring the logger here and importing it wherever logging is required, every logger will be named [utils.py] (that's what the __name__ method does. You're essentially only creating one instance of a logger and using it everywhere

A better method is to use a get_logger function or a loggerfactory that can be used to return a new instance of a logger where ever needed, with some custom attributes that can be supplied by the caller such as name and default log_level

Comment (kushal-ti) [reply restapi/utils.py:13]: Also, while the current organization is good, it's more preferable to separate out separate utils in separate files than having all utils in a single file. For ex. have a utils folder in which you have logger_utils.py and log_processor_utils.py

**Review by kushal-ti — APPROVED** [review]
Good stuff, probably the best one I've seen so far. You even fixed some stuff I wasn't aware of, like removing all the dead code from that one serializer class. Left a few comments, go over them , but i'm approving this and closing the PR

File: restapi/utils.py (hunk 1 above)
Comment (abhishek-dhanraj-TI) [reply restapi/utils.py:13]: Okay. I'll make sure to follow this next time.
Thank you for your explanation.

**Pull request closed without merging**